src/app/utils/utils.js

```diff
@@ -813,7 +813,24 @@ function aeonUrl(item) {
   return AeonUrl.toString();
 }
 
+// transform bib id to have lower case prefix (b, hb, cb, pb) and trim check digit
+function standardizeBibId(bibId) {
+  const nypl = bibId.match(/^([bB])(\d{8})[\dxX]?$/)
```

Comment:
will the last digit always be an "x"? Why is that?

Reply:
And it's expected to only be once, right? Asking because this works
`http://local.nypl.org:3001/research/research-catalog/bib/b17418167x` but this does not
`http://local.nypl.org:3001/research/research-catalog/bib/b17418167xx` and I'm not sure if that's required.

Reply:
It won't always be an x. [\dxX] will match on one of x, X or a digit. It could be an x because the check digit is base 11, I think, x being the digit that comes after 9. 

An nypl bibid with 10 digits is not a valid id, and we're not concerned with fixing incorrectly entered ids, only in transforming correct ones. Technically we should be checking the check digit to make sure it's valid, but I didn't do that in this implementation